connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -228,7 +228,9 @@ final class DataFrameWriter[T] private[sql] (ds: Dataset[T]) {
 
     // Set path or table
     f(builder)
-    require(builder.hasPath != builder.hasTable) // Only one can be set
+
+    // Cannot both be set, hasPath or hasTable or nothing
+    require(builder.hasPath || builder.hasTable || (!builder.hasPath && !builder.hasTable))
```

Comment:
It seems to pass for both set case?

`require(!(builder.hasPath && builder.hasTable))`?